### PR TITLE
not for merge : Labelconnectcomponent

### PR DESCRIPTION
**Pull request by StRigaud**
Apparent issue with nonzero_minimum_box kernel not working when used with flag dimension {1,1,1}
Already spotted by @haesleinhuepf

### CODE REVIEW

File: include/CLE.h [hunk 1]
```diff
@@ -83,6 +83,8 @@ class CLE
     void SmallerOrEqual(Buffer&, Buffer&, Buffer&);
     void SmallerConstant(Buffer&, Buffer&, float);
     void SmallerOrEqualConstant(Buffer&, Buffer&, float);
+    void LabelConnectComponent(Buffer&, Buffer&);
```
Comment (haesleinhuepf) [inline include/CLE.h:86]: If you name it "ConnectedComponentsLabelingBox", then the code generator would work with it.

**Comment by haesleinhuepf** [comment]
With flag dimension 1x1x2 it should work. There is a glitch inside the create method, I assume. If depth==1, it creates a 2D image. Which is different on Java and Python side... I think we should strive for having 3D kernels only .... long term, not for tomorrow , ;-)

**Comment by StRigaud** [comment]
set dimension to 1x1x2 works.
kernel seems to run fine but CL_INVALID_VALUE is raised when pull() is called on output buffer

**Comment by haesleinhuepf** [comment]
Yeah, sometimes the driver dies one step after running a defective kernel...

**Comment by StRigaud** [comment]
the CL_INVALID_VALUE was my fault. I change a parameter of ReadBuffer() breaking all pull() function ... fixed now.
All compile and run but the return buffer value of LabelConnectComponentBoxKernel is full 0 whatever the input.

File: src/cleLabelConnectComponentBoxKernel.cpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    CopyKernel copy(this->gpu);
+    if (iteration_count % 2 == 0)
+    {    
+        copy.SetInput(temp1);
+    }
+    else
+    {
+        copy.SetInput(temp1);
```
Comment (haesleinhuepf) [inline src/cleLabelConnectComponentBoxKernel.cpp:109]: I think this should copy from temp2

File: src/cleLabelConnectComponentBoxKernel.cpp [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        flag_value = ReadBuffer<int>(flag_mem, 2, this->gpu)[0];
+
+        SetKernel set(this->gpu);
+        set.SetInput(flag);
+        set.SetValue(0);
+        set.Execute();
+
+        iteration_count++;
+    }
+
```
Comment (haesleinhuepf) [inline src/cleLabelConnectComponentBoxKernel.cpp:101]: By printing out `temp1` here, I can see that it contains pixel values > 0. They are wrong, but anyway: That would mean, copy doesn't work....

```
    int* data = ReadBuffer<int>(tmp1_mem, 100, this->gpu);
    // Verify output
    for (size_t i = 0; i < 100; i++)
    {
        if (i % 10 == 0) {
            std::cout << std::endl;
        }
        std::cout << data[i] << " ";
    }
```

File: src/cleLabelConnectComponentBoxKernel.cpp (hunk 2 above)
Comment (StRigaud) [reply src/cleLabelConnectComponentBoxKernel.cpp:109]: fixed. It was late :)

File: src/cleLabelConnectComponentBoxKernel.cpp (hunk 3 above)
Comment (StRigaud) [reply src/cleLabelConnectComponentBoxKernel.cpp:101]: fix copy, tag were in reversed order, added a test also

Now print a non empty output but no consistency in between labels though

**Comment by StRigaud** [comment]
Branch not relevant anymore. 
More advance implementation is in progress (wait for it).

**Pull request closed without merging**